grafana-plugin/src/components/WithConfirm/WithConfirm.tsx

```diff
@@ -1,18 +1,22 @@
 import React, { ReactElement, useCallback, useState } from 'react';
 
-import { ConfirmModal } from '@grafana/ui';
+import { ConfirmModal, ConfirmModalProps } from '@grafana/ui';
 
-interface WithConfirmProps {
+type WithConfirmProps = Partial<ConfirmModalProps> & {
```

Comment:
I think `interface WithConfirmProps extends ConfirmModalProps` is better, in your current implementation `confirmText` exists in both `Partial<ConfirmModalProps>` and `{confirmText?: string;}`

Reply:
Good call 👍  I think `title` and `confirmText` are unnecessary here
```diff
type WithConfirmProps = Partial<ConfirmModalProps> & {
  children: ReactElement;
--  title?: string;
--  confirmText?: string;
  disabled?: boolean;
};
```